ssl/ssl_lib.c

```diff
@@ -3884,6 +3883,17 @@ int SSL_get_error(const SSL *s, int i)
         (s->s3.warn_alert == SSL_AD_CLOSE_NOTIFY))
         return SSL_ERROR_ZERO_RETURN;
 
+    /*
+     * Make things return SSL_ERROR_SYSCALL when doing SSL_do_handshake etc,
+     * where we do encode the error
+     */
+    if ((l = ERR_peek_error()) != 0) {
+        if (ERR_GET_LIB(l) == ERR_LIB_SYS)
+            return SSL_ERROR_SYSCALL;
+        else
+            return SSL_ERROR_SSL;
+    }
```

Comment:
Why is this necessary? Any error set by libssl should have been caught by `ossl_statem_in_error` above

Reply:
ossl_statem_in_error could not handle the error from application protocol,this kept for compatibility.